File: Leetcode/leetcode-1394-FindLuckyIntegerInAnArray-[array]-[easy]/Solution.java
class Solution {
    public int findLucky(int[] arr) {
        int largest = -1;
        HashMap<Integer, Integer> freq = new HashMap<>();
        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            if (entry.getKey() == entry.getValue()) {
                largest = Math.max(largest, entry.getKey());
            }
        }

        return largest;
    }
}

File: Leetcode/leetcode-1394-FindLuckyIntegerInAnArray-[array]-[easy]/Solution.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        largest = -1
        freq = dict({})
        for num in arr:
            freq.update({ num: freq.get(num, 0) + 1 })
        
        for k, v in freq.items():
            if (k == v):
                largest = max(largest, k)
        
        return largest